### PR TITLE
Adding ProductID to ApplePayDetails

### PR DESCRIPTION
**Pull request by niharika88**
### Description
Update ApplePay responses to properly receive Product_ID

### Checklist

* [x] I've read the [contribution guidelines](CONTRIBUTING.md)
* [x] I've added Unit Tests if necessary.
* [x] I've checked whether additional documentation is needed.
* [x] I've ensured any personally identifying information is handled with the necessary care.
* [x] I've checked if my implementation doesn't break other features.

### CODE REVIEW

**Review by 0sc — APPROVED** [review]

**Comment by 0sc** [comment]
:shipit: 

Approved with Zappr  Approved with Zappr

**Comment by ozairasim** [comment]
⚡ 

Approved with Zappr  Approved with Zappr

**Comment by ozairasim** [comment]
⚡ 

Approved with Zappr  Approved with Zappr

**Comment by Anirudhlanka89** [comment]
+1

Approved with Zappr

File: transaction_apple_pay_details_integration_test.go [hunk 1]
```diff
@@ -49,6 +49,9 @@ func TestTransactionApplePayDetails(t *testing.T) {
 	if tx.ApplePayDetails.CardholderName == "" {
 		t.Fatal("Expected ApplePayDetails to have CardholderName set")
 	}
+	if tx.ApplePayDetails.ProductID == "" {
```
Comment (muh-hamada) [inline transaction_apple_pay_details_integration_test.go:52]: I think this values exists only if certain conditions were met (sending level 2 and 3 data)
So maybe we can mark it as optional?

**Comment by 0sc** [comment]
:shipit: 


Approved with Zappr  Approved with Zappr

File: transaction_apple_pay_details_integration_test.go (hunk 1 above)
Comment (niharika88) [reply transaction_apple_pay_details_integration_test.go:52]: ok

**Comment by muh-hamada** [comment]
👍

Approved with Zappr  Approved with Zappr  Approved with Zappr

**Comment by ozairasim** [comment]
👍 

Approved with Zappr  Approved with Zappr

File: transaction_apple_pay_details_integration_test.go [hunk 2]
```diff
@@ -49,6 +49,9 @@ func TestTransactionApplePayDetails(t *testing.T) {
...
+		t.Errorf("Expected ApplePayDetails to have ProductID set")
```
Comment (muh-hamada) [inline transaction_apple_pay_details_integration_test.go]: I think it should be a warning

Comment (ozairasim) [reply transaction_apple_pay_details_integration_test.go]: I see that it is being set as "Unknown" when we don't send L2 and L3 data if that makes any difference

**Review by niharika88 — COMMENTED** [review]
Added changes

File: transaction_apple_pay_details_integration_test.go (hunk 2 above)
Comment (niharika88) [reply transaction_apple_pay_details_integration_test.go]: I have changed it to LogF

**Comment by ozairasim** [comment]
⚡ 

Approved with Zappr  Approved with Zappr

**Comment by muh-hamada** [comment]
👍

Approved with Zappr  Approved with Zappr  Approved with Zappr

**Comment by 0sc** [comment]
+1

Approved with Zappr  Approved with Zappr

**Pull request merged**